### PR TITLE
Add trusted paths UI to Security tab in preferences

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

- Adds the skeleton UI for adding trusted paths to security tab in preferences.

Screen Shot 2022-05-02 at 3 37 51 PM

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

- Add trusted paths UI to Security tab in preferences

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

**Comment by zeusongit** [comment]
Using the same resource strings as package paths for now as both were same

File: src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        /// <summary>
+        /// Indicate the path to the custom packages folder
+        /// </summary>
+        public string Path { get; set; }
+    }
+
+    public class TrustedPathViewModel : ViewModelBase
+    {
+        public ObservableCollection<string> TrustedLocations { get; private set; }
+        [Obsolete("SelectedIndex is no longer referenced, do not use.")]
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs:33]: Is the obsolete message intended?

File: src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            SaveSettingCommand = new DelegateCommand(CommitChanges);
+        }
+
+        private int ConvertPathToIndex(object path)
+        {
+            if(path is int pint)
+            {
+                return pint;
+            }
+            return TrustedLocations.IndexOf(path as string);
```
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/Menu/TrustedPathViewModel.cs:78]: Do we really need to handle both strings and integers when managing CRUD operations ?
I would suggest we only handle strings, basically only keep line 78

**Comment by pinzart90** [comment]
@zeusongit please show a screen shot of the latest UI

**Comment by zeusongit** [comment]
> @zeusongit please show a screen shot of the latest UI

It is attached to the PR, there is no folder attached, therefore the list of paths is not visible.

**Comment by pinzart90** [comment]
> > @zeusongit please show a screen shot of the latest UI
> 
> It is attached to the PR, there is no folder attached, therefore the list of paths is not visible.

Can you mock the folder paths so that we can see the UI controls (hardcode in local code should be fine for getting the screenshot)

**Comment by zeusongit** [comment]
@pinzart90 I tried hard coding the path, but it seems there are other things that break and are needed for this to work, i may have to implement some part of the API to get it to work, like add binding triggers etc. If you are concerned about the look and UI, it is exactly the same as in Package manager add path UI.

**Comment by pinzart90** [comment]
> @pinzart90 I tried hard coding the path, but it seems there are other things that break and are needed for this to work, i may have to implement some part of the API to get it to work, like add binding triggers etc. If you are concerned about the look and UI, it is exactly the same as in Package manager add path UI.

How do you know it is exactly the same ? Did you get it to work/show in Dynamo UI in earlier stages of this PR ?
Or are going by how the code looks like ?

**Comment by zeusongit** [comment]
Screenshot updated.

**Review by pinzart90 — APPROVED** [review]
LGTM

**Comment by zeusongit** [comment]
Merging, the failing check is due to artifactory outage

**Pull request merged**